Refactor functional api vectorized jacobian to use batched grad parameter

soulitzer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #65566
* #65564
* #65565

This doesn't simplify vectorized jacobian computation, but is good to consolidate logic and helps us to test the logic

Differential Revision: [D31236257](https://our.internmc.facebook.com/intern/diff/D31236257)

soulitzer committed:
Refactor functional api vectorized jacobian to use batched grad parameter

[ghstack-poisoned]

soulitzer committed:
Update on "Refactor functional api vectorized jacobian to use batched grad parameter"




[ghstack-poisoned]

soulitzer committed:
Update on "Refactor functional api vectorized jacobian to use batched grad parameter"


This doesn't simplify vectorized jacobian computation, but is good to consolidate logic and helps us to test the logic

[ghstack-poisoned]

soulitzer committed:
Update on "Refactor functional api vectorized jacobian to use batched grad parameter"


This doesn't simplify vectorized jacobian computation, but is good to consolidate logic and helps us to test the logic

[ghstack-poisoned]

soulitzer commented:
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31236257).

soulitzer committed:
Update on "Refactor functional api vectorized jacobian to use batched grad parameter"


This doesn't simplify vectorized jacobian computation, but is good to consolidate logic and helps us to test the logic

Differential Revision: [D31236257](https://our.internmc.facebook.com/intern/diff/D31236257)

[ghstack-poisoned]

soulitzer commented:
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31236257).

albanD reviewed (approved):
LGTM

soulitzer committed:
Update on "Refactor functional api vectorized jacobian to use batched grad parameter"


This doesn't simplify vectorized jacobian computation, but is good to consolidate logic and helps us to test the logic

Differential Revision: [D31236257](https://our.internmc.facebook.com/intern/diff/D31236257)

[ghstack-poisoned]

zou3519 commented on torch/autograd/functional.py:
```diff
@@ -427,14 +427,10 @@ def jacobian(func, inputs, create_graph=False, strict=False, vectorize=False):
-            use `torch._C._debug_only_display_vmap_fallback_warnings(True)`
-            to show any performance warnings and file us issues if
-            warnings exist for your use case. Defaults to ``False``.
+            ``True``, we perform only a single ``autograd.grad`` call with
+            ``batched_grad=True`` which uses the vmap prototype feature.
+            See :func:`torch.autograd.grad`'s ``batched_grad`` parameter for
+            more information.
```
There should be a larger warning about the performance cliffs here

soulitzer committed:
Update on "Refactor functional api vectorized jacobian to use batched grad parameter"


This doesn't simplify vectorized jacobian computation, but is good to consolidate logic and helps us to test the logic

Differential Revision: [D31236257](https://our.internmc.facebook.com/intern/diff/D31236257)

[ghstack-poisoned]

soulitzer committed:
Update on "Refactor functional api vectorized jacobian to use batched grad parameter"


This doesn't simplify vectorized jacobian computation, but is good to consolidate logic and helps us to test the logic

Differential Revision: [D31236257](https://our.internmc.facebook.com/intern/diff/D31236257)

[ghstack-poisoned]

soulitzer committed:
Update on "Refactor functional api vectorized jacobian to use batched grad parameter"


This doesn't simplify vectorized jacobian computation, but is good to consolidate logic and helps us to test the logic

Differential Revision: [D31236257](https://our.internmc.facebook.com/intern/diff/D31236257)

[ghstack-poisoned]

soulitzer committed:
Update on "Refactor functional api vectorized jacobian to use batched grad parameter"


This doesn't simplify vectorized jacobian computation, but is good to consolidate logic and helps us to test the logic

Differential Revision: [D31236257](https://our.internmc.facebook.com/intern/diff/D31236257)

[ghstack-poisoned]

soulitzer committed:
Update on "Refactor functional api vectorized jacobian to use batched grad parameter"


This doesn't simplify vectorized jacobian computation, but is good to consolidate logic and helps us to test the logic

Differential Revision: [D31236257](https://our.internmc.facebook.com/intern/diff/D31236257)

[ghstack-poisoned]

soulitzer commented:
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31236257).

soulitzer commented:
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31236257).

soulitzer committed:
Update on "Refactor functional api vectorized jacobian to use batched grad parameter"


This doesn't simplify vectorized jacobian computation, but is good to consolidate logic and helps us to test the logic

Differential Revision: [D31236257](https://our.internmc.facebook.com/intern/diff/D31236257)

[ghstack-poisoned]

soulitzer commented:
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31236257).

soulitzer committed:
Update on "Refactor functional api vectorized jacobian to use batched grad parameter"


This doesn't simplify vectorized jacobian computation, but is good to consolidate logic and helps us to test the logic

Differential Revision: [D31236257](https://our.internmc.facebook.com/intern/diff/D31236257)

[ghstack-poisoned]

soulitzer commented:
@soulitzer has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D31236257).
